src/transport/AdminPairingTable.h

```diff
@@ -343,6 +343,17 @@ class DLL_EXPORT AdminPairingTable
     ConstAdminIterator begin() const { return cbegin(); }
     ConstAdminIterator end() const { return cend(); }
 
+    uint8_t GetFabricIndex(AdminPairingInfo * fabric) const
```

Comment:
Value of zero is invalid. Furthermore, the rules for fabric index allocation are such that the fabric index is monotonically increasing until wrap-around back to 1.

The method used here couples strongly to the underlying storage mechanism.

It can remain like this in this PR, as it would be an easy fix for a subsequent PR, but the solution is to add a FabricIndex field to each AdminPairingInfo entry, and have an allocator with counter persistence for the FabricIndex generation.